packages/hashi/src/Bloom/BloomRunner.js

```diff
@@ -66,67 +132,37 @@ export default class BloomRunner {
   initBloom() {
     try {
       this.loaded();
-      this.iframe.src = `../bloom/bloomplayer.htm?url=${this.contentUrl}&distributionUrl=${this.distributionUrl}&metaJsonUrl=${this.metaUrl}&independent=false`;
+      const options = new URLSearchParams({
+        url: this.contentUrl,
+        distributionUrl: this.distributionUrl,
+        metaJsonUrl: this.metaUrl,
+        independent: false,
+        hideFullScreenButton: true,
+        initiallyShowAppBar: true,
+        allowToggleAppBar: false,
+      });
+
+      this.iframe.src = `../bloom/bloomplayer.htm?${options.toString()}`;
     } catch (e) {
       this.errored(e);
     }
   }
 
-  processContent() {
```

Comment:
is this

> Migrates all the bloom player specific code that handles mapping audio files into the bloom player code, use the extensibility of the kolibri-zip module to allow this

?

Reply:
Yes - the `processContent` method was doing post-hoc processing because the mapping logic wasn't being setup for the zip file in the first place. All of what was happening here is now happening inside the BloomDOMMapper class that is being defined in here, in order to make it happen while we are extracting files from the zip file.